common/src/main/java/com/microsoft/identity/common/internal/cache/SharedPreferencesFileManager.java

```diff
@@ -137,44 +137,64 @@ public final String getSharedPreferencesFileName() {
     public final void putString(
             final String key,
             final String value) {
+        final String methodTag = TAG + ":putString";
+
         synchronized (cacheLock) {
             if (value != null) {
                 fileCache.put(key, value);
             } else {
                 fileCache.remove(key);
             }
+
             final SharedPreferences.Editor editor = mSharedPreferences.edit();
 
             if (null == mEncryptionManager || StringUtil.isNullOrEmpty(value)) {
-                editor.putString(key, value);
-            } else {
-                final String encryptedValue = encrypt(value);
-                editor.putString(key, encryptedValue);
+                editor.putString(key, value).apply();
+                return;
             }
 
-            editor.apply();
+            // NOTE: Maybe the right behavior is that we shouldn't write NULL whenever encryption fails.
+            // However, it's possible that we're relying on this behavior in PROD (due to KeyStore failure)
+            // Not persisting the key in memory means getAll() or getAllFilteredByKey() will not work.
```

Comment:
Just want to double check if I understand correctly, getAll() would not work correctly (with your proposed change) because, if we do not override it with null, the old (non null) value would be persisted and it would end up returning the old value.. whereas the encryption failed and it should not have returned the old value?

Reply:
GetAll() relies on the key in storage.

Let's say if i have an empty storage. encryption fails. 
The current logic would have value in the in-memory cache, and the key in both in-memory and storage.

If i call getAll(), i would still be able to get the key (from storage) and use that to retrieve value (in-memory).


If I don't persist (key, null), then there would be no persisted key, and the  persisted value wouldn't be retrieved.

Reply:
I'll reword the comment to make it clearer

Reply:
To clarify, the word `key` here means a `name` in the map? (rather than an encryption key)

Reply:
yeah